tests/e2e/lifecycletest/pvc_snap_clone/test_workload_pvc_snapshot.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+import time
+import pytest
+
+from subprocess import CalledProcessError
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version, E2ETest, tier1
+)
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility import templating
+from ocs_ci.ocs.exceptions import CommandFailed
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@skipif_ocs_version('<4.6')
+@pytest.mark.polarion_id("OCS-2302")
+class TestPvcSnapshotOfWorkloads(E2ETest):
+    """
+    Tests to verify PVC snapshot feature for pgsql workloads
+    """
+
+    def test_pvc_snapshot(self, pgsql_factory_fixture, snapshot_factory, snapshot_restore_factory):
+        """
+        1. Deploy PGSQL workload
+        2. Take a snapshot of the pgsql PVC.
+        3. Create a new PVC out of that snapshot or restore snapshot
+        4. Attach a new pgsql pod to it.
+        5. Create pgbench benchmark to new pgsql pod
+
+        """
+
+        # Deploy PGSQL workload
+        log.info("Deploying pgsql workloads")
+        pgsql = pgsql_factory_fixture(replicas=3, clients=3, transactions=600)
+
+        # Get postgres pvc list obj
+        postgres_pvcs_obj = pgsql.get_postgres_pvc()
+
+        # Take a snapshot
+        log.info("Creating snapshot of all postgres PVCs")
+        snapshots = []
+        for pvc_obj in postgres_pvcs_obj:
+            log.info(
+                f"Creating snapshot of PVC {pvc_obj.name}"
+            )
+            snap_obj = snapshot_factory(pvc_obj=pvc_obj, snapshot_name=f"{pvc_obj.name}-snap")
+            snapshots.append(snap_obj)
+        log.info("Snapshots creation completed and in Ready state")
+
+        # Create PVCs out of the snapshots
+        log.info("Creating new PVCs from snapshots")
+        restore_pvc_objs = []
+        for snapshot in snapshots:
+            log.info(f"Creating a PVC from snapshot {snapshot.name}")
+            restore_pvc_obj = snapshot_restore_factory(
+                snapshot_obj=snapshot,
+                restore_pvc_name=f"{snapshot.name}-restored",
+                size='10Gi',
```

Comment:
it is better to take this value from the created PVC (original) and not use a constant.

Reply:
agreed, removed and directly gets from Parent PVC